src/state/slices/tradeInputSlice/selectors.ts

```diff
@@ -99,9 +101,32 @@ export const selectFirstHopSellAccountId = createSelector(
   },
 )
 
-// selects the account ID we're selling from for the other hops
-// for posterity, every hop always sells from the same account as the first hop
-export const selectSecondHopSellAccountId = selectFirstHopSellAccountId
+// if multi-hop, selects the account ID we're selling from for fee asset of the last hop and its account number
+// else, selects the AccountId from the first hop
+export const selectSecondHopSellAccountId = createSelector(
+  selectAccountNumberParamFromFilter,
+  selectAccountIdByAccountNumberAndChainId,
+  selectChainIdParamFromFilter,
+  // selectIsActiveQuoteMultiHop,
+  selectFirstHopSellAccountId,
+  (
+    sellAssetAccountNumber,
+    sellAssetAccountIds,
+    sellAssetChainId,
+    // isMultiHopTrade,
+    firstHopSellAccountId,
+  ) => {
+    // TODO(gomes): fix circular deps here
```

Comment:
This will obviously break for non-multi-hop quotes as it stands, we need to do what the TODO says to do

![do it](https://media0.giphy.com/media/v1.Y2lkPTc5MGI3NjExc3M1NXo1am9wNWpwMTh1bGZiamw0OTJodXZpc2UyMXI2ZHJyaXoxMyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/UqZ4imFIoljlr5O2sM/giphy.gif)